### PR TITLE
I20 trajectory 1

### PR DESCRIPTION
**Pull request by yshiyi**
A class of input functions which can generate four particular trajectories such as spiral, nudges, swerves and corner has been defined. 
Four examples have been designed to present the output of these four input functions.
User may check a specific example by choosing the accosiated index number at the beginning of the program.
User may also give an arbitrary initial condition to the motion model right after choosing the index number.
In addition, a new motion model module has been constructed so that users may generate any motions by modifying this module.

### CODE REVIEW

**Comment by chutsu** [comment]
Is `i20_Trajectory` not the one you are editing? if not please remove it?

**Comment by chutsu** [comment]
Several points to consider:
- Maybe put this in the `matlab_simulation/7-Control/` folder?
- Since you are asking the user for the input trajectory why not put each of them into their own functions/file?
- Plot code is almost the same across all examples, create a general plot function?
- Remove the `robottrak_backup` file, its kind of redundant since we can use git to git it back

File: matlab_simulation/code v1.0/ME597-7-Trajectories/robottraj.m [hunk 1]
```diff
@@ -1,85 +1,133 @@
...
+    
+elseif index == 4
+    % Corner
+    v = 2*ones(size(T));
+    w = zeros(size(T));
+    c = floor(length(w)/8);
+    w(2*c+1:5*c) = -2/5;
+
+    for t=1:length(T)
+        xddot(:,t) = [v(t)*cos(xd(3,t));
```
Comment (chutsu) [inline matlab_simulation/code v1.0/ME597-7-Trajectories/robottraj.m]: Can you not have used `motion_model()` function here? as well in lines 35, 57 and 80?

File: matlab_simulation/code v1.0/ME597-7-Trajectories/motion_model.m [hunk 2]
```diff
@@ -0,0 +1,13 @@
+function xddot = motion_model(t, T, xd)
```
Comment (chutsu) [inline matlab_simulation/code v1.0/ME597-7-Trajectories/motion_model.m]: `v` and `w` both should be input variables to the motion model, also consider giving this function a better name other than `motion_model()`, I assume this is for a two wheeled robot?

File: matlab_simulation/code v1.0/ME597-7-Trajectories/robottraj.m [hunk 3]
```diff
@@ -1,85 +1,133 @@
...
-xd(:,1)= x0;
-v = 2*ones(size(T));
-w = zeros(size(T));
-c = floor(length(w)/8);
+if index == 1
+    % Spiral
+    v = sin(0.2*T);
+    w = ones(size(T));
+    
+    for t=1:length(T)
```
Comment (chutsu) [inline matlab_simulation/code v1.0/ME597-7-Trajectories/robottraj.m]: The iteration of the motion model is very much the same across other examples, factor this for loop out? Same goes to all other examples?

**Comment by neginlashkari** [comment]
This is a neat code to generate various trajectories for robots. This code allows the user to choose the function and its initial values. However, I think it would be better to instead of generally asking for initial conditions, let the user know what are the role of each initial variables in final trajectory. 
This way user can generate whatever trajectory he needs by entering initial conditions.

**Comment by neginlashkari** [comment]
Also, when I choose "user defined motion model" as input trajectory, it gives me spiral function. I understood that we should write our desired function in "motion_mode.m". I was wondering if we can also make the user able to write his desired function in command window and get the results (instead of changing the mfile).

**Comment by chutsu** [comment]
In response to allowing user to input desired function in the command window, I personally feel that would just add complexity to the simple example. You're already running matlab, its rather trivial to edit a file in the same directory, not to mention its rather annoying to type and correct mistakes in the command window multiple times.

**Comment by yshiyi** [comment]
This code only focused on generating different trajectories. Path planning is not considered.
Six different trajectories: spiral, nudges, swerves, corner, lane change, obstacle avoid are generated.
In addition, users can define their own trajectory.
Users are also allowed to give any initial condition except number 6, obstacle avoidance.
Each individual motion has been generated in its own function. 
Users can select which trajectory to run and give any initial condition as well.
'motion_model.m' is the general motion model for all six pre-defined motions and also can be edited in each individual function.
Users are allowed to define and edit their own motion model in 'user_define_motion_model.m'. 
'input_check.m' can check whether the input index is valid or not.

**Pull request merged**